### PR TITLE
fix(global): use authorization_code instead of id_token for user login

### PR DESCRIPTION
**Pull request by Samy-33**

### CODE REVIEW

File: api/api/auth/views.py [hunk 1]
```diff
@@ -20,9 +22,19 @@ def post(self, request):
...
+                auth_response = google_client.exchange_auth_code_for_token(code)
+            except Exception:
+                logging.error('Error while exchanging auth code for token')
+                return Response(
+                    status=403,
+                    data={'message': 'Invalid authorization code. Try logging in again.'})
+            try:
+                # If the verification is done too soon, an error is thrown.
+                # Following 2 seconds pause is used to avoid that error.
+                time.sleep(2)
```
Comment (Aksh-Bansal-dev) [inline api/api/auth/views.py:36]: why do we need to sleep for 2 seconds

File: ui/src/api/googleOauth.ts [hunk 2]
```diff
@@ -28,31 +22,28 @@ export const openGoogleOAuthPage = () => {
...
+ * @param authorizationCode - Authorization code returned by google
  * @param redirectToHome - callback function to redirect to home
  */
 export const convertAccessToken = async (
-  googleAccessToken: string,
+  authorizationCode: string,
   redirectToHome: () => void,
   failRedirect: () => void
 ) => {
+  console.log(authorizationCode);
```
Comment (Aksh-Bansal-dev) [inline ui/src/api/googleOauth.ts:33]: Remove console.log

File: ui/.env.template [hunk 3]
```diff
@@ -1,5 +1,3 @@
 VITE_GOOGLE_CLIENT_ID=
-VITE_API_CLIENT_ID=
-VITE_API_CLIENT_SECRET=
```
Comment (Aksh-Bansal-dev) [inline ui/.env.template:3]: Also, remove these from `constants.ts`

**Review by Aksh-Bansal-dev — CHANGES_REQUESTED** [review]
Maybe `global` scope would be better in the commit message.

File: api/api/auth/views.py (hunk 1 above)
Comment (Samy-33) [reply api/api/auth/views.py:36]: Explained in the comments

File: ui/.env.template (hunk 3 above)
Comment (Samy-33) [reply ui/.env.template:3]: Already removed.

File: ui/src/api/googleOauth.ts (hunk 2 above)
Comment (Samy-33) [reply ui/src/api/googleOauth.ts:33]: This should probably be caught in linting test. Can you figure out if there's a way?

Comment (Aksh-Bansal-dev) [reply ui/src/api/googleOauth.ts:33]: Yeah, we can add this in `.eslintrc.js`.
```javascript
...
  rules: {
    'no-console': 2,
...
```

Comment (Samy-33) [reply ui/src/api/googleOauth.ts:33]: Create an issue for this.

**Review by Aksh-Bansal-dev — APPROVED** [review]
LGTM.

**Pull request merged**